pkg/apis/core/validation/namespacedcloudprofile.go

```diff
@@ -85,3 +93,48 @@ func validateParent(parent core.CloudProfileReference, fldPath *field.Path) fiel
 
 	return allErrs
 }
+
+func validateOnlyExpirationsKubernetesVersions(versions []core.ExpirableVersion, fldPath *field.Path) field.ErrorList {
```

Comment:
What does "only expirations" mean in this context? 👀

Reply:
I split up checking the values as they are provided (i.e. in the plain config) and the context check to check for conflicts and other restrictions. That's why this function for NamespacedCloudProfiles ended up checking only the expiration dates.
I'll adapt the name again together with the other suggested refactorings.